### PR TITLE
Version updates (since DR2) and resulting build fixes

### PR DESCRIPTION
**Pull request by jsenko**
I've updated two dependency versions, and the spark one (result of an update of spark in jdg-spark)
caused a compilation failure that i needed to fix quickly so I could proceed with the release.

@gustavonalle Please check if the fixes are correct and complete.

### CODE REVIEW

File: spark/pom.xml [hunk 1]
```diff
@@ -43,7 +43,7 @@
         <version.org.infinispan>8.5.0-redhat-SNAPSHOT</version.org.infinispan>
         <version.spark.connector>7.2.0-redhat-SNAPSHOT</version.spark.connector>
 
-        <version.spark>2.0.0</version.spark>
+        <version.spark>2.1.0</version.spark>
```
Comment (gustavocoding) [inline spark/pom.xml:46]: We should target Spark 2.2.0

File: spark/pom.xml [hunk 2]
```diff
@@ -43,7 +43,7 @@
         <version.org.infinispan>8.5.0-redhat-SNAPSHOT</version.org.infinispan>
         <version.spark.connector>7.2.0-redhat-SNAPSHOT</version.spark.connector>
```
Comment (gustavocoding) [inline spark/pom.xml:44]: This version is wrong, it should be ```0.5.0-redhat-SNAPSHOT```

File: spark/spark-temperature-analysis/src/main/scala/org/infinispan/quickstart/spark/TemperatureAnalysis.scala [hunk 3]
```diff
@@ -62,7 +64,7 @@ object TemperatureAnalysis {
       val configIn = new Properties
       configIn.put("infinispan.rdd.cacheName", inputDataGridCache)
       configIn.put("infinispan.client.hotrod.server_list", dataGridServer)
-      val ispnStream = new InfinispanInputDStream[String, Double](ssc, StorageLevel.MEMORY_ONLY, configIn)
+      val ispnStream = new InfinispanInputDStream[String, Double](ssc, StorageLevel.MEMORY_ONLY, org.infinispan.spark.config.ConnectorConfiguration(configIn.asInstanceOf[java.util.Map[String, String]].asScala.toMap))
```
Comment (gustavocoding) [inline spark/spark-temperature-analysis/src/main/scala/org/infinispan/quickstart/spark/TemperatureAnalysis.scala:67]: The configuration via ```Properties``` has been [replaced](http://blog.infinispan.org/2017/04/infinispan-spark-connector-05-released.html).
So this snippet is better written as:
```scala
import org.infinispan.spark.config.ConnectorConfiguration;
...
val configIn = new ConnectorConfiguration().setCacheName("inputDataGridCache").setServerList(dataGridServer)
val ispnStream = new InfinispanInputDStream[String, Double](ssc, StorageLevel.MEMORY_ONLY, configIn) 
```

File: spark/spark-temperature-analysis/src/main/scala/org/infinispan/quickstart/spark/TemperatureAnalysis.scala [hunk 4]
```diff
@@ -85,7 +87,7 @@ object TemperatureAnalysis {
       val configOut = new Properties
       configOut.put("infinispan.rdd.cacheName", outputDataGridCache)
       configOut.put("infinispan.client.hotrod.server_list", dataGridServer)
-      avgTemperatures.writeToInfinispan(configOut)
+      avgTemperatures.writeToInfinispan(org.infinispan.spark.config.ConnectorConfiguration(configOut.asInstanceOf[java.util.Map[String, String]].asScala.toMap))
```
Comment (gustavocoding) [inline spark/spark-temperature-analysis/src/main/scala/org/infinispan/quickstart/spark/TemperatureAnalysis.scala:90]: The same applies to this change: create a ```ConnectorConfiguration``` and pass it directly to the ```writeToInfinispan``` method.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by jsenko** [comment]
See https://github.com/jboss-developer/jboss-jdg-quickstarts/pull/180

**Pull request closed without merging**